orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineExpressionFunctionProvider.java

```diff
@@ -0,0 +1,97 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.front50.pipeline;
+
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.pipeline.expressions.ExpressionFunctionProvider;
+import com.netflix.spinnaker.orca.pipeline.expressions.SpelHelperFunctionException;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+@Component
+public class PipelineExpressionFunctionProvider implements ExpressionFunctionProvider {
+  /** Since expression evaluation happens in a static function we can't get access to front50 there
+   * Since this is a component it will be a singleton and hence safe to save the front50
+   * reference here
+   */
+  private static Front50Service front50Service;
+
+  @Autowired
+  PipelineExpressionFunctionProvider(Front50Service front50Service) {
+    PipelineExpressionFunctionProvider.front50Service = front50Service;
+  }
+
+
+  @Nullable
+  @Override
+  public String getNamespace() {
+    return null;
+  }
+
+  @NotNull
+  @Override
+  public Collection<FunctionDefinition> getFunctions() {
+    return Arrays.asList(
+      new FunctionDefinition("pipeline", Arrays.asList(
+        new FunctionParameter(
+          Execution.class, "execution", "The execution containing the currently executing stage"
+        ),
+        new FunctionParameter(
+          String.class, "pipelineName", "A valid stage reference identifier"
+        )
+      ))
+    );
+  }
+
+  /**
+   * Function to convert pipeline name to pipeline ID (within current application)
+   *
+   * @param execution the current execution
+   * @param pipelineName name of the pipeline to lookup
+   * @return the id of the pipeline or null if pipeline not found
+   */
+  public static String pipeline(Execution execution, String pipelineName) {
+    if ((pipelineName == null) || (pipelineName.isEmpty())) {
+      throw new SpelHelperFunctionException("pipelineName must be specified for function #pipeline");
+    }
+
+    if (front50Service == null) {
+      throw new SpelHelperFunctionException("front50 service is missing. It's required when using #pipeline function");
+    }
+
+    String currentApplication = execution.getApplication();
+
+    Map<String, Object> pipeline = front50Service.getPipelines(currentApplication)
+      .stream()
+      .filter(p -> pipelineName.equals(p.getOrDefault("name", null)))
+      .findFirst()
+      .orElse(null);
+
+    if (pipeline != null) {
+      return (String)pipeline.getOrDefault("id", null);
+    }
+
+    return null;
```

Comment:
Perhaps worth throwing an exception that the pipeline couldn't be found rather than returning null?

Reply:
If you went the exception route ... `orElseThrow()` and then you can avoid the `pipeline != null` check.

Reply:
yeah, i like it, thanks! i think it makes debugging expressions failures easier